src/components/PasswordInput.vue

```diff
@@ -0,0 +1,61 @@
+<template>
+  <label class="base-input">
+    <span class="base-input__label">{{ label }}</span>
+    <div class="input-wrapper">
+      <input
+        class="base-input__input"
+        :type="type"
+        placeholder="Enter your password"
+        :value="value"
+        @input="$emit('input', $event.target.value)"
+      />
+      <span class="input-wrapper__icon"> </span>
+    </div>
+  </label>
+</template>
+
+<script>
+export default {
+  name: "PasswordInput",
+  props: {
+    label: {
+      type: String,
+      default: "",
+    },
+    value: {
+      type: String,
+      default: "",
+    },
+    type: {
+      type: String,
+      default: "password",
+    },
+  },
+
+  // methods: {
+  //   showPassword() {
+  //     this.$emit("show-password");
+  //   },
+  // },
```

Comment:
I see that you were trying to make it dynamic but you gave up?

Let me suggest something: 
This component name is `PasswordInput` so I think it is totally fine to assume that input `type` is handled inside the component (no need for a prop `type`). You can define `type` attribute for the `<input>` here inside the component without emitting anything.